conjure-runtime/src/service/retry.rs

```diff
@@ -179,13 +186,27 @@ where
                     _ => return Err(error),
                 }
 
+                #[allow(clippy::if_same_then_else)] // conditionals get too crazy if combined
```

Comment:
We can factor out the cases to helper methods and combine the cases.

Reply:
I think this is all going to get rewritten entirely for #58, so I think it might be best to just leave it until that change.